lib/routes/account.js

```diff
@@ -298,6 +276,37 @@ module.exports = function (
           }
         }
 
+        function sendVerifyCode () {
+          if (! account.emailVerified) {
+            return mailer.sendVerifyCode(account, account.emailCode, {
+              service: form.service || query.service,
+              redirectTo: form.redirectTo,
+              resume: form.resume,
+              acceptLanguage: request.app.acceptLanguage
+            })
+            .then(function () {
+              // only create reminder if sendVerifyCode succeeds
+              verificationReminder.create({
+                uid: account.uid.toString('hex')
+              }).catch(function (err) {
+                log.error({ op: 'Account.verificationReminder.create', err: err })
+              })
+            })
+            .catch(function (err) {
+              // If the email was rejected, delete the account.
+              if (err.errno !== error.ERRNO.EMAIL_REJECTED) {
```

Comment:
Heh, yeah, the missing piece here is figuring out how to get enough error info out of the mailer, that we know for sure this was a delivery failure rather than some other transient error.